scripts/wsrep_sst_clone.sh

```diff
@@ -0,0 +1,1113 @@
+        CLONE_SSL_CERT="NULL"
+        CLONE_SSL_KEY="NULL"
+        CLONE_SSL_CA="NULL"
+    else
+        CLEANUP_CLONE_PLUGIN=""
+        CLONE_SSL_CERT=`$MYSQL_ACLIENT -e "SELECT @@clone_ssl_cert"`
+        CLONE_SSL_KEY=`$MYSQL_ACLIENT -e "SELECT @@clone_ssl_key"`
+        CLONE_SSL_CA=`$MYSQL_ACLIENT -e "SELECT @@clone_ssl_ca"`
+    fi
+
+    local CLIENT_SSL_CERT=$(parse_cnf sst ssl_cert "")
+    local CLIENT_SSL_KEY=$(parse_cnf sst ssl_key "")
+    local CLIENT_SSL_CA=$(parse_cnf sst ssl_ca "")
+    local CLIENT_SSL_MODE=$(parse_cnf sst ssl_mode "")
+
+    if [ -z "$CLIENT_SSL_CERT" -o -z "$CLIENT_SSL_KEY" ]
+    then
+        CLIENT_SSL_CERT=$(parse_cnf client ssl_cert "")
+        CLIENT_SSL_KEY=$(parse_cnf client ssl_key "")
+        CLIENT_SSL_CA=$(parse_cnf client ssl_ca "")
+        CLIENT_SSL_MODE=$(parse_cnf client ssl_mode "")
+    fi
+
+    local SERVER_SSL_CERT=`$MYSQL_ACLIENT -e "SELECT @@ssl_cert"`
+    local SERVER_SSL_KEY=`$MYSQL_ACLIENT -e "SELECT @@ssl_key"`
+    local SERVER_SSL_CA=`$MYSQL_ACLIENT -e "SELECT @@ssl_ca"`
+
+    [ "$SERVER_SSL_CERT" = "NULL" ] && SERVER_SSL_CERT=
+    [ "$SERVER_SSL_KEY" = "NULL" ] && SERVER_SSL_KEY=
+    [ "$SERVER_SSL_CA" = "NULL" ] && SERVER_SSL_CA=
+
+    if [ "$CLONE_SSL_CERT" = "NULL" -o "$CLONE_SSL_KEY" = "NULL" ]
+    then
+        wsrep_log_info "CLONE SSL not configured. Checking general MySQL SSL settings."
+
+        if [ "$ROLE" = "donor" ]
+        then
+            CLONE_SSL_CERT=$SERVER_SSL_CERT
+            CLONE_SSL_KEY=$SERVER_SSL_KEY
+            CLONE_SSL_CA=$SERVER_SSL_CA
+        else
+            CLONE_SSL_CERT=$CLIENT_SSL_CERT
+            CLONE_SSL_KEY=$CLIENT_SSL_KEY
+            CLONE_SSL_CA=$CLIENT_SSL_CA
+        fi
+
+        if [ -n "$CLONE_SSL_CERT" -a -n "$CLONE_SSL_KEY" ]
+        then
+            wsrep_log_info "Using SSL configuration from MySQL Server."
+            $MYSQL_ACLIENT -e "SET GLOBAL clone_ssl_cert='$CLONE_SSL_CERT'"
+            $MYSQL_ACLIENT -e "SET GLOBAL clone_ssl_key='$CLONE_SSL_KEY'"
+            $MYSQL_ACLIENT -e "SET GLOBAL clone_ssl_ca='$CLONE_SSL_CA'"
+            CLEANUP_CLONE_SSL="yes"
+        fi
+    else
+        if [ -n "$CLONE_SSL_CERT" -a -n "$CLONE_SSL_KEY" ]
+        then
+            wsrep_log_info "CLONE SSL already configured. Using it."
+        else
+            wsrep_log_info "CLONE SSL is explicitly empty: @@clone_ssl_cert='$CLONE_SSL_CERT', @@clone_ssl_key='$CLONE_SSL_KEY'"
+        fi
+        CLEANUP_CLONE_SSL=
+    fi
+
+    CLIENT_SSL_OPTIONS=""
+    if [ -n "$CLONE_SSL_CERT" -a -n "$CLONE_SSL_KEY" ]
+    then
+        wsrep_log_info "Server SSL settings on $ROLE: CERT=$SERVER_SSL_CERT, KEY=$SERVER_SSL_KEY, CA=$SERVER_SSL_CA"
+        wsrep_log_info "Client SSL settings on $ROLE: CERT=$CLIENT_SSL_CERT, KEY=$CLIENT_SSL_KEY, CA=$CLIENT_SSL_CA"
+        wsrep_log_info "CLONE SSL settings on $ROLE: CERT=$CLONE_SSL_CERT, KEY=$CLONE_SSL_KEY, CA=$CLONE_SSL_CA"
+        REQUIRE_SSL="REQUIRE SSL"
```

Comment:
Here, we set the REQURIE_SSL variable to `REQUIRE SSL`, but it gets overwritten in line number 428? Is this expected? 

This is what we will get in the error log even when both server and client have the ssl certificates in place.
```
2024-11-22T09:05:44.965420Z 0 [Note] [MY-000000] [WSREP-SST] Installing CLONE plugin
2024-11-22T09:05:45.086583Z 0 [Note] [MY-000000] [WSREP-SST] CLONE SSL not configured. Checking general MySQL SSL settings.
2024-11-22T09:05:45.086607Z 0 [Note] [MY-000000] [WSREP-SST] Using SSL configuration from MySQL Server.
2024-11-22T09:05:45.112218Z 0 [Note] [MY-000000] [WSREP-SST] Server SSL settings on recipient: CERT=/home/venki/work/pxc/merge/80/mysql-test/std_data/server-cert.pem, KEY=/home/venki/work/pxc/merge/80/mysql-test/std_data/server-key.pem, CA=/home/venki/work/pxc/merge/80/mysql-test/std_data/cacert.pem
2024-11-22T09:05:45.112236Z 0 [Note] [MY-000000] [WSREP-SST] Client SSL settings on recipient: CERT=/home/venki/work/pxc/merge/80/mysql-test/std_data/server-cert.pem, KEY=/home/venki/work/pxc/merge/80/mysql-test/std_data/server-key.pem, CA=/home/venki/work/pxc/merge/80/mysql-test/std_data/cacert.pem
2024-11-22T09:05:45.112256Z 0 [Note] [MY-000000] [WSREP-SST] CLONE SSL settings on recipient: CERT=/home/venki/work/pxc/merge/80/mysql-test/std_data/server-cert.pem, KEY=/home/venki/work/pxc/merge/80/mysql-test/std_data/server-key.pem, CA=/home/venki/work/pxc/merge/80/mysql-test/std_data/cacert.pem
2024-11-22T09:05:45.112321Z 0 [Note] [MY-000000] [WSREP-SST] SSL Not supported yet.

```

Reply:
Yes I have intentionally disabled SSL until we are ok with the global process. 
Then will focus on SSL. Have SSL between the Donor/joiner is not a big problem if the certificates are correctly installed. More cumbersome is to deal with the temp instance doing the clone action. But most of the logic is already in the script, I just ignore it in favour of the clone mechanics.

Reply:
SSL enabled and working.